core/lib/web3_decl/src/error.rs

```diff
@@ -42,3 +43,65 @@ pub enum Web3Error {
     #[error("Tree API is not available")]
     TreeApiUnavailable,
 }
+
+use std::{collections::HashMap, error, fmt, fmt::Debug};
+
+#[derive(Debug)]
+pub struct RpcErrorWithDetails {
+    pub inner_error: RpcError,
+    pub method: String,
```

Comment:
Are there use cases when `method` is not known in compile time (i.e., cannot be a `&'static str`)? Same question regarding `args` keys.